test/NumberCollectorState.cpp

```diff
@@ -5,32 +5,36 @@
 
 #include "TankControllerLib.h"
 
+unittest(Edge_Case) {
```

Comment:
Thank you for adding a test for the broken code. It demonstrates good discipline!